opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -1,60 +1,122 @@
-use opentelemetry::logs::{LogRecord, Logger, LoggerProvider, Severity};
+use opentelemetry::{
+    logs::{AnyValue, LogRecord, Logger, LoggerProvider, Severity},
+    Key,
+};
 use std::borrow::Cow;
-use tracing_core::Level;
+use tracing_core::{Level, Metadata};
+use tracing_log::NormalizeEvent;
 use tracing_subscriber::Layer;
 
 const INSTRUMENTATION_LIBRARY_NAME: &str = "opentelemetry-appender-tracing";
+// Start by "log."
+const LOG_METADATA_ATTRIBUTES: &[&str] = &[
+    "name",
+    "target",
+    "module.path",
+    "file.path",
+    "file.name",
+    "file.line",
+    "module_path",
+    "file",
+    "line",
+];
 
 /// Visitor to record the fields from the event record.
-struct EventVisitor<'a> {
-    log_record: &'a mut LogRecord,
+#[derive(Default)]
+struct EventVisitor {
+    log_record_attributes: Vec<(Key, AnyValue)>,
+    log_record_body: Option<AnyValue>,
 }
 
-impl<'a> tracing::field::Visit for EventVisitor<'a> {
+fn is_metadata(field: &str) -> bool {
+    if let Some(log_field) = field.strip_prefix("log.") {
+        for log_metadata_field in LOG_METADATA_ATTRIBUTES {
+            if *log_metadata_field == log_field {
+                return true;
+            }
+        }
+    }
+    false
+}
+
+fn get_filename(filepath: &str) -> &str {
+    if let Some((_, filename)) = filepath.rsplit_once('/') {
+        return filename;
+    }
+    if let Some((_, filename)) = filepath.rsplit_once('\\') {
+        return filename;
+    }
+    filepath
+}
+
+impl EventVisitor {
+    fn visit_metadata(&mut self, meta: &Metadata) {
+        self.log_record_attributes
+            .push(("log.name".into(), meta.name().into()));
+        self.log_record_attributes
+            .push(("log.target".into(), meta.target().to_owned().into()));
```

Comment:
Given this is based on [experimental ](https://opentelemetry.io/docs/specs/semconv/general/logs/)conventions, this may block us from releasing stable version of the appender. In other words, we'll be dependent on this semantic convention becoming stable, to release this crate as stable. I am not sure if that is a risk we should take, given the historical delays in stabilizing sem. conventions!

Open to ideas how to navigate this space....

Reply:
Good point. We can also have these under the experimental feature flag, till the semantic conventions are stable.

Reply:
I'm good with the feature flag. I can add it.

Reply:
@cijothomas Do you agree with the feature flag solution?

Reply:
Yes, feature-flag is a good way to manage the risk.